pkg/serviceaccounttoken/secret.go

```diff
@@ -35,13 +34,13 @@ const (
 type secretLister func(namespace string, selector labels.Selector) ([]*corev1.Secret, error)
 
 // EnsureSecretForServiceAccount gets or creates a service account token Secret for the provided Service Account.
-func EnsureSecretForServiceAccount(ctx context.Context, secretsCache corecontrollers.SecretCache, clientSet kubernetes.Interface, sa *corev1.ServiceAccount) (*corev1.Secret, error) {
```

Comment:
I'm not a fan of passing such a "powerful" client, it makes dependencies harder to track, at the cost of one extra parameter and adding noise to this PR, but I think it's worth it.